stock_move_source_relocate/models/stock_move.py

```diff
@@ -35,37 +36,30 @@ def _apply_source_relocate(self):
             )
 
     def _apply_source_relocate_rule(self, relocation, reserved_availability, roundings):
-        relocated = self.env["stock.move"].browse()
-
+        self.ensure_one()
         rounding = roundings[self]
-        if not reserved_availability[self]:
+        qty_reserved = reserved_availability[self]
+        if float_compare(qty_reserved, 0, precision_rounding=rounding) <= 0:
```

Comment:
Could the qty_reserved really < 0 ?

Reply:
I shouldn't be lower than 0. 
https://github.com/odoo/odoo/blob/7c105dc37bc0e8d740b86af3f9e88ddeb3c98f54/addons/stock/models/stock_move_line.py#L294